ewatercycle/models/wflow.py

```diff
@@ -0,0 +1,157 @@
+import shutil
+import time
+from os import PathLike
+from pathlib import Path
+from typing import Any, Iterable, Optional, Tuple
+
+import numpy as np
+import xarray as xr
+from cftime import num2date
+from grpc4bmi.bmi_client_docker import BmiClientDocker
+from grpc4bmi.bmi_client_singularity import BmiClientSingularity
+
+from ewatercycle.models.abstract import AbstractModel
+from ewatercycle.parametersetdb.config import CaseConfigParser
+
+# from ewatercycle import CFG
+
+# mock CFG until the CFG PR is merged.
+CFG = {
+    "container_engine": "docker",
+    "singularity_images.wflow": "ewatercycle-wflow-grpc4bmi.sif",
+    "docker_images.wflow": "ewatercycle/wflow-grpc4bmi:latest",
+    "scratch_dir": "./"
+}
+
+
+class Wflow(AbstractModel):
+    """eWaterCycle implementation of WFLOW hydrological model.
+
+    Attributes
+        bmi (Bmi): GRPC4BMI Basic Modeling Interface object
+    """
+    def setup( # type: ignore
+            self,
+            cfg_dir: PathLike,
+            cfg_file: PathLike,
+            forcing_data: Optional[PathLike] = None,
+            **kwargs) -> Tuple[PathLike, PathLike]:
+        """Start the model inside a container and return a valid config file.
+
+        Args:
+
+            - cfg_dir: path to a valid wflow model spec.
+
+            - cfg_file: path to the configuration file, typically somethig like
+            `wflow_sbm.ini`. This overwrites any existing files in cfg_dir with
+            the same filename.
+
+            - forcing_data (optional): path to meteorological forcing data (.nc)
+            file. If not given, it should already be in the cfg_dir.
+
+            - **kwargs (optional, dict): any settings in the cfg_file that you want
+            to overwrite programmatically. Should be passed as a dict, e.g.
+            `run = {"starttime": "1995-01-31 00:00:00 GMT"}` where run is the
+            section in which the starttime option may be found.
+
+        Returns:
+            Path to config file and config dir
+        """
+        self._setup_cfg_dir(cfg_dir=cfg_dir)
+        self._setup_cfg_file(cfg_file=cfg_file, **kwargs)
+        self._setup_forcing(forcing_data=forcing_data)
+        self._start_container()
+
+        return self.cfg_dir, self.cfg_file
```

Comment:
Can dir and file be swapped? See https://pymt.readthedocs.io/en/latest/usage.html#model-setup
```suggestion
        return self.cfg_file, self.cfg_dir
```

Reply:
Good spot, I swapped them